app/src/examples/ModifyExample.tsx

```diff
@@ -0,0 +1,46 @@
+import { Button, StyleSheet, View } from 'react-native';
+
+import React from 'react';
+import {
+  runOnUI,
+  useDerivedValue,
+  useSharedValue,
+} from 'react-native-reanimated';
+
+export default function ModifyExample() {
+  const sv = useSharedValue([1]);
+
+  const handleModify = () => {
+    sv.modify((value) => {
+      'worklet';
+      value.push(value.length + 1);
+      return value;
+    });
+  };
+
+  useDerivedValue(() => {
+    console.log('useDerivedValue', sv.value);
+  });
+
+  const handleRead = () => {
+    console.log('JS', sv.value);
```

Comment:
I assume that we expect in this case for these to be different. We should add a comment about that so that it doesn't confuse people testing this in the future.